pkg/build/strategies/layered/layered.go

```diff
@@ -161,16 +162,10 @@ func (builder *Layered) Build(config *api.Config) (*api.Result, error) {
 	// hex image id ref is not kosher with the docker API; so we remove the ID piece, and then
 	// construct the new image name
 	var newBuilderImage string
-	if len(dockerImageReference.ID) == 0 {
-		newBuilderImage = fmt.Sprintf("%s-%d", builder.config.BuilderImage, time.Now().UnixNano())
+	if _, hasDigest := namedReference.(reference.Digested); hasDigest {
+		newBuilderImage = fmt.Sprintf("%s:s2i-layered-%d", namedReference.Name(), time.Now().UnixNano())
 	} else {
-		if len(dockerImageReference.Registry) > 0 {
-			newBuilderImage = fmt.Sprintf("%s/", dockerImageReference.Registry)
-		}
-		if len(dockerImageReference.Namespace) > 0 {
-			newBuilderImage = fmt.Sprintf("%s%s/", newBuilderImage, dockerImageReference.Namespace)
-		}
-		newBuilderImage = fmt.Sprintf("%s%s:s2i-layered-%d", newBuilderImage, dockerImageReference.Name, time.Now().UnixNano())
+		newBuilderImage = fmt.Sprintf("%s-%d", builder.config.BuilderImage, time.Now().UnixNano())
```

Comment:
can you see a reason not to include "s2i-layered" in the name here like we do in the other path?